apm-agent-plugins/apm-api-plugin/src/main/java/co/elastic/apm/agent/pluginapi/LegacySpanInstrumentation.java

```diff
@@ -104,7 +104,14 @@ public static class AdviceClass {
             @Advice.OnMethodExit(inline = false)
             public static Object doCreateSpan(@Advice.FieldValue(value = "span", typing = Assigner.Typing.DYNAMIC) Object span) {
                 if (span instanceof Span<?>) {
-                    return ((Span<?>) span).createSpan();
+                    Span<?> spanTyped = (Span<?>) span;
+                    //TODO: make the public API work on contexts instead of spans, otherwise baggage might be lost
```

Comment:
I'm not sure what you meant here, is that something that will be fixed by your next PR when adding baggage ?

Reply:
I'll remove the "TODO" and replace it with a more clear comment as this limitation will stay.
The limitation is if you start a span on the public API via `ElasticAPM.currentSpan().startSpan()` then you will loose baggage updates which happened after the `currentSpan()` was started.
Imo it is not worth investing in lifting this limitation, as this can only occur if you are actively mixing Otel (for accessing baggage) and our own public API. The solution is to just stay with Otel.

I'll also document this limitation on the Baggage issue.